tron/core/actionrun.py

```diff
@@ -653,6 +653,47 @@ def submit_command(self):
         mesos_cluster.submit(task)
         return task
 
+    def recover(self):
+        if self.mesos_task_id is None:
+            log.error(f'No task ID, cannot recover {self}')
+            return
+
+        okay_to_recover = self.machine.transition('running')
+        if not okay_to_recover:
+            log.error(
+                f'Unable to transition {self} from {self.machine.state}'
+                'to running for recovery'
+            )
+            return
+
+        log.info(f'Recovering Mesos run {self}')
+        self.exit_status = None
+        self.end_time = None
+
+        serializer = filehandler.OutputStreamSerializer(self.output_path)
+        mesos_cluster = MesosClusterRepository.get_cluster()
+        task = mesos_cluster.create_task(
```

Comment:
Are we sure that mesos cluster would be able to reregister the framework at this point? Do we need to worry about the race condition between reregister and task recovery? Or am I missing any safeguard to prevent this somewhere?

Reply:
The .check method just tells task_processing to keep track of the task ID, the actual requests to Mesos don't happen until the framework is ready. So no race condition.